tensorboard/__init__.py

```diff
@@ -0,0 +1,263 @@
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorboard import plugins
+
+class FileWriter(tf.summary.FileWriter):
+  """Writes `Summary` protocol buffers to event files.
+
+  The `FileWriter` class provides a mechanism to create an event file in a
+  given directory and add summaries and events to it. The class updates the
+  file contents asynchronously. This allows a training program to call methods
+  to add data to the file directly from the training loop, without slowing down
+  training.
+  """
+
+  def __init__(self,
+               logdir,
+               graph=None,
+               max_queue=10,
+               flush_secs=120,
+               filename_suffix=None):
+    """Creates a `FileWriter` and an event file.
+
+    On construction the summary writer creates a new event file in `logdir`.
+    This event file will contain `Event` protocol buffers constructed when you
+    call one of the following functions: `add_summary()`, `add_session_log()`,
+    `add_event()`, or `add_graph()`.
+
+    If you pass a `Graph` to the constructor it is added to
+    the event file. (This is equivalent to calling `add_graph()` later).
+
+    TensorBoard will pick the graph from the file and display it graphically so
+    you can interactively explore the graph you built. You will usually pass
+    the graph from the session in which you launched it:
+
+    ```python
+    ...create a graph...
+    # Launch the graph in a session.
+    sess = tf.Session()
+    # Create a summary writer, add the 'graph' to the event file.
+    writer = tb.FileWriter(<some-directory>, sess.graph)
+    ```
+
+    The other arguments to the constructor control the asynchronous writes to
+    the event file:
+
+    *  `flush_secs`: How often, in seconds, to flush the added summaries
+       and events to disk.
+    *  `max_queue`: Maximum number of summaries or events pending to be
+       written to disk before one of the 'add' calls block.
+
+    Args:
+      logdir: A string. Directory where event file will be written.
+      graph: A `Graph` object, such as `sess.graph`.
+      max_queue: Integer. Size of the queue for pending events and summaries.
+      flush_secs: Number. How often, in seconds, to flush the
+        pending events and summaries to disk.
+      filename_suffix: A string. Every event file's name is suffixed with
+        `suffix`.
+    """
+    super(FileWriter, self).__init__(logdir,
+                                     graph=graph,
+                                     max_queue=max_queue,
+                                     flush_secs=flush_secs,
+                                     filename_suffix=filename_suffix)
+
+
+  def get_logdir(self):
+    """Returns the directory where event file will be written."""
+    return super(FileWriter, self).get_logdir()
```

Comment:
Linter is complaining that these are all "useless `super` delegations"; it wants you to remove these methods entirely. Feel free to disable that warning each time that it occurs (or for this class, if you prefer).